libraries/ms-common/test/test_can.c

```diff
@@ -275,3 +278,49 @@ void test_can_default(void) {
   TEST_ASSERT_EQUAL(msg.msg_id, rx_msg.msg_id);
   TEST_ASSERT_EQUAL(msg.data, rx_msg.data);
 }
+
+#ifdef X86
+void test_can_x86_tx(void) {
+  volatile CanMessage rx_msg = { 0 };
+  CanMessage msg = {
+    .msg_id = 0xF,              //
+    .type = CAN_MSG_TYPE_DATA,  //
+    .data = 0x1,                //
+    .dlc = 1,                   //
+  };
+
+  can_register_rx_handler(0xF, prv_rx_callback, &rx_msg);
+  s_rx_cb_count = 0;
+
+  for (uint8_t i = 0x0; i < 0xA; i++) {
+    msg.data = i;
+    can_transmit(&msg, NULL);
+  }
+
+  Event e = { 0 };
+  uint8_t tx_msg_count = 0;
+  uint8_t rx_msg_count = 0;
+  uint8_t loop_count = 0;
+
+  while (rx_msg_count < 10 || tx_msg_count < 10) {
+    event_process(&e);
+    can_process_event(&e);
+    if (e.id == TEST_CAN_EVENT_TX) {
+      tx_msg_count++;
+    }
+    delay_ms(10);
+    if (e.id == TEST_CAN_EVENT_RX) {
+      rx_msg_count++;
+    }
+    loop_count++;
+    if (loop_count > 30) {
+      TEST_FAIL();
+    }
```

Comment:
You can just do `TEST_ASSERT(loop_count <= 30)` (or 3 * the constant you define)